src/m365/commands/status.ts

```diff
@@ -14,19 +14,33 @@ class StatusCommand extends Command {
   }
 
   public commandAction(logger: Logger, args: any, cb: (err?: any) => void): void {
+
     if (auth.service.connected) {
-      if (this.debug) {
-        logger.logToStderr({
-          connectedAs: accessToken.getUserNameFromAccessToken(auth.service.accessTokens[auth.defaultResource].accessToken),
-          authType: AuthType[auth.service.authType],
-          accessTokens: JSON.stringify(auth.service.accessTokens, null, 2)
-        });
-      }
-      else {
-        logger.log({
-          connectedAs: accessToken.getUserNameFromAccessToken(auth.service.accessTokens[auth.defaultResource].accessToken)
+      auth
+        .ensureAccessToken(auth.defaultResource, logger, this.debug)
+        .then((): void => {
+          if (this.debug) {
+            logger.logToStderr({
+              connectedAs: accessToken.getUserNameFromAccessToken(auth.service.accessTokens[auth.defaultResource].accessToken),
+              authType: AuthType[auth.service.authType],
+              accessTokens: JSON.stringify(auth.service.accessTokens, null, 2)
+            });
+          }
+          else {
+            logger.log({
+              connectedAs: accessToken.getUserNameFromAccessToken(auth.service.accessTokens[auth.defaultResource].accessToken)
+            });
+          }
+          cb();
+        }, (rej: Error): void => {
+          if (this.debug) {
+            logger.logToStderr(rej);
+          }
+
+          logger.log('Your login has expired. Sign in again to continue.');
```

Comment:
Instead of logging a separate message, let's add this error to the one returned by the failed token retrieval and pass them both as a single string to CommandError on line 42.

Reply:
thanks @waldekmastykarz , i have made the change. do let me know if I did not interpret this in correct way.